registry/storage/driver/filesystem/driver.go

```diff
@@ -224,6 +224,41 @@ func (d *driver) Move(ctx context.Context, sourcePath string, destPath string) e
 	}
 
 	err := os.Rename(source, dest)
+	// os.Rename will not work across devices, in that case, try to copy then delete instead
+	if err != nil {
+		err = d.copyFile(source, dest)
+		// If we just moved an uploaded blob, the PurgeUploads loop probably removed it already,
+		// therefore we must check if the source file still exists before trying to delete it
+		if _, err := os.Stat(source); err == nil {
+			if err = os.RemoveAll(source); err != nil {
+				return err
+			}
+		}
+	}
+	return err
+}
+
+// copy a file to another
+func (d *driver) copyFile(src, dst string) error {
+	in, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer in.Close()
+	out, err := os.Create(dst)
```

Comment:
Please allocate a tempfile here. Otherwise, we risk writing to the same file in two different processes. This is not optional.